lib/Humming-Bird/Core.rakumod

```diff
@@ -408,23 +420,49 @@ sub group(@routes, @middlewares) is export {
     .(@middlewares) for @routes;
 }
 
-sub routes(--> Hash) is export {
+multi sub advice(--> List:D) is export {
+    @ADVICE.clone;
+}
+
+multi sub advice(@advice) is export {
+    @ADVICE.push: |@advice;
+}
+
+multi sub advice(&advice) is export {
+    @ADVICE.push: &advice;
+}
+
+sub error($type, &handler) is export {
+    %ERROR{$type.^name} = &handler;
+}
+
+sub routes(--> Hash:D) is export {
     %ROUTES.clone;
 }
 
-sub listen(Int $port) is export {
-    my HTTPServer $server = HTTPServer.new(port => $port);
-    $server.listen(-> $raw_request {
-        my Request $request = Request.encode($raw_request);
-        my Bool $keep_alive = False;
-        with $request.headers<Connection> {
-            $keep_alive = $request.headers<Connection>.lc eq 'keep-alive';
+sub handle($raw-request) {
+    my Request $request = Request.encode($raw-request);
+    my Bool $keep-alive = False;
+    my &advice = [o] @ADVICE; # Advice are Response --> Response
+    with $request.headers<Connection> {
+        $keep-alive = $request.headers<Connection>.lc eq 'keep-alive';
```

Comment:
`with` sets the topic so I'd consider not typing `$request.headers<Connection>` again.